libs/blocks/merch/merch.js

```diff
@@ -179,10 +180,21 @@ export async function fetchEntitlements() {
   return fetchEntitlements.promise;
 }
 
+async function loadDefaultPriceLiterals() {
+  return fetch(DEFAULT_PRICE_LITERALS_PATH);
+}
+
 export async function fetchLiterals(url) {
   fetchLiterals.promise = fetchLiterals.promise ?? new Promise((resolve) => {
     fetch(url)
-      .then((response) => response.json().then(({ data }) => resolve(data)));
+      .catch(() => loadDefaultPriceLiterals())
+      .then(async (response) => {
+        if (response.ok) {
+          return response.json().then(({ data }) => resolve(data));
+        }
+        const defaultPriceLiterals = await loadDefaultPriceLiterals();
+        return defaultPriceLiterals.json().then(({ data }) => resolve(data));
+      });
```

Comment:
We usually write the `.then` block first and afterwards the `.catch`
I'd also chain the promises
```suggestion
      .then((response) => {
        if (response.ok) return response.json();
        return loadDefaultPriceLiterals()
          .then((defaultPriceLiterals) => defaultPriceLiterals.json());
      })
      .then(({ data }) => resolve(data));
      .catch(() => loadDefaultPriceLiterals())
```

Reply:
@mokimo are you sure? :) In method `fetchCheckoutLinkConfigs()` we have `fetch().catch().then();`
I chained promises as you describe but `catch()` needs to stay where it is since I load default json in `catch()` which is then processed in the following `then()` methods. I tried to move `catch()` to the last place but then nothing works of course. It seems that the order of handlers matters with `fetch()`.

Reply:
Are you sure you run into the `.catch` block? It's very rare to get into that.

https://developer.mozilla.org/en-US/docs/Web/API/Window/fetch
> A fetch() promise only rejects when the request fails, for example, because of a badly-formed request URL or a network error. A fetch() promise does not reject if the server responds with HTTP status codes that indicate errors (404, 504, etc.). Instead, a then() handler must check the [Response.ok](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok) and/or [Response.status](https://developer.mozilla.org/en-US/docs/Web/API/Response/status) properties.

Reply:
Sorry @mokimo we are changing the context for this PR like crazy. The code we are talking about here is now removed. I updated the PR description with new ideas.